vortex-file/src/v2/segments/reader.rs

```diff
@@ -0,0 +1,22 @@
+//! The segment reader provides an async interface to layouts for resolving individual segments.
+
+use bytes::Bytes;
+use futures::channel::oneshot;
+use vortex_array::aliases::hash_map::HashMap;
+use vortex_error::VortexResult;
+use vortex_layout::segments::SegmentId;
+
+type OneShot<T> = (oneshot::Sender<T>, oneshot::Receiver<T>);
+
+pub struct SegmentCache {
```

Comment:
why do you have two of them?

Reply:
Oh just experimenting, ignore everything in vortex-file, only really care about vortex-layout changes